components/channel_header/components/header_icon_wrapper.tsx

```diff
@@ -146,4 +157,4 @@ const HeaderIconWrapper: React.FC<Props> = (props: Props) => {
     );
 };
 
-export default HeaderIconWrapper;
+export default memo(HeaderIconWrapper);
```

Comment:
Not sure @michelengelen. Most of the code is primarily coming from this PR:
https://github.com/mattermost/mattermost-webapp/pull/6138

It's a community PR, and I was just fixing a UI bug and conflicts on it so it could get in. Looks like the whole PR needs quite some work before it can be done.